### PR TITLE
Update limiter project

### PR DESCRIPTION
**Pull request by tommbendall**
This is a correction to the `ThetaLimiter` which wasn't working as it should. This limiter is used with the Embedded DG advection scheme, which finishes with a projection step.

This pull request makes a simple change to the projection step, so that it uses the recovery operator rather than a Galerkin projection (this was what we intended to do originally).

I also took the opportunity to update the limiter test so that it was no longer separated into vertical and horizontal parts -- it's now a rotational test with both horizontal and vertical motion.

### CODE REVIEW

File: gusto/advection.py [hunk 1]
```diff
@@ -92,6 +92,9 @@ def _setup(self, state, field, options):
             self.Projector = Projector(self.xdg_out, self.x_projected,
                                        solver_parameters=parameters)
 
+        if options.name == "embedded_dg" and self.limiter is not None:
```
Comment (jshipton) [inline gusto/advection.py]: does this mean that self.Projector is defined but not used when `options.name=="embedded_dg"` and there is no limiter?

Comment (tommbendall) [reply gusto/advection.py]: Yes you're right -- I could put this under an `else`

File: gusto/limiters.py [hunk 2]
```diff
@@ -154,7 +150,6 @@ def apply(self, field):
         self.copy_vertex_values(field)
         self.vertex_limiter.apply(self.theta_hat)
         self.copy_vertex_values_back(field)
-        self.remap_to_embedded_space(field)
```
Comment (tommbendall) [inline gusto/limiters.py]: Since I am not using this function I should also delete it

File: gusto/advection.py (hunk 1 above)
Comment (jshipton) [reply gusto/advection.py]: maybe also find a name that works for both `self.Projector` and `self.x_recoverer` then you can simplify the code in lines 144-147

**Pull request merged**